### PR TITLE
DNF plugin fixes and debugging

### PR DESCRIPTION
**Pull request by kmittman**
- Do not block kernel updates if DKMS stream kmod package is installed
 - Retrieve k_corepkg hawkey package object rather than string
 - Catch errors with try-except blocks
 - Expose plugin as DNF command (for debugging)
 - Print verbose output when running as a command

### CODE REVIEW

File: nvidia-dnf.py [hunk 1]
```diff
@@ -56,6 +57,13 @@ def sack(self):
         available_kernels = sack.query().available().filter(name = KERNEL_PKG_NAME)
         driver = installed_drivers[0]
 
+        if installed_modules:
+            string_modules = ' '.join([str(elem) for elem in installed_modules])
+
+        # DKMS stream enabled
+        if installed_modules and 'dkms' in string_modules:
```
Comment (tbaederr) [inline nvidia-dnf.py:94]: Seems you can just do this in the first `if`? Or is `string_modules` needed later?

File: nvidia-dnf.py [hunk 2]
```diff
@@ -70,19 +78,30 @@ def sack(self):
...
+                    sack.add_excludes([kernelpkg])
+                    sack.add_excludes([k_corepkg])
+                    exclude = True
+                except Exception as error:
+                    print('WARNING: kernel exclude error', error)
+
+                if exclude == True:
+                    print('NOTE: Skipping kernel installation since no NVIDIA driver kernel module package ' + \
+                        str(kmod_pkg_name) + ' for kernel ' + str(kernelpkg) + ' and driver ' + \
+                        str(driver) + ' could be found')
```
Comment (tbaederr) [inline nvidia-dnf.py:132]: Can this print also go into the `try` block at the end and replace the `exclude` variable? Or is that not how Python's try/except works? I found it confusing that `exclude` is only defined in the `try` block but I guess that works in Pyhton.

**Comment by kmittman** [comment]
Hi Timm,
I'll have to tackle those improvements for the next version, as v1.9 released today.

File: nvidia-dnf.py (hunk 1 above)
Comment (kmittman) [reply nvidia-dnf.py:94]: I want it to `return` after it has printed all of the debug information

File: nvidia-dnf.py (hunk 2 above)
Comment (kmittman) [reply nvidia-dnf.py:132]: Fixed in #4

**Pull request merged**